File: 13.Linked_List/javascript/1.CountNode.js
// Define the Node class
class Node {
    constructor(data) {
        this.data = data;
        this.next = null;
    }
}

// Manually create the linked list
const head = new Node(10);
const second = new Node(20);
const third = new Node(30);

// Link the nodes
head.next = second;
second.next = third;

// Function to count the number of nodes
function countNodes(head) {
    let count = 0;
    let current = head;
    while (current !== null) {
        count++;
        current = current.next;
    }
    return count;
}

// Call the function and display the count
console.log("Number of nodes:", countNodes(head));

File: 13.Linked_List/python/1.CountNode.py
# Define the Node class (already given)
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

# Manually create the linked list
head = Node(10)
second = Node(20)
third = Node(30)

# Link the nodes
head.next = second
second.next = third

# Count the number of nodes
def count_nodes(head):
    count = 0
    current = head
    while current:
        count += 1
        current = current.next
    return count

# Call the function and display the count
print("Number of nodes:", count_nodes(head))
